File: 703. Kth Largest Element in a Stream
/Solution.cpp
class KthLargest {
public:
    int K;
    priority_queue<int, vector<int>, greater<int>> pq;
    KthLargest(int k, vector<int>& nums) {
        K = k;
        for (auto& n : nums) {
            pq.push(n);
            if (pq.size() == K + 1) pq.pop();
        }
    }
    
    int add(int val) {
        pq.push(val);
        if (pq.size() == K + 1) pq.pop();
        return pq.top();
    }
};

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest* obj = new KthLargest(k, nums);
 * int param_1 = obj->add(val);
 */

File: 703. Kth Largest Element in a Stream
/Solution.java
class KthLargest {
    int K;
    PriorityQueue<Integer> pq;
    public KthLargest(int k, int[] nums) {
        pq = new PriorityQueue<>();
        K = k;
        for (int n : nums) {
            pq.offer(n);
            if (pq.size() == K + 1) pq.poll();
        }
    }
    
    public int add(int val) {
        pq.offer(val);
        if (pq.size() == K + 1) pq.poll();
        return pq.peek();
    }
}

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest obj = new KthLargest(k, nums);
 * int param_1 = obj.add(val);
 */
